### PR TITLE
feat(rest) : Adding or Modifying fields to project summaryadministration API

### PR DESCRIPTION
**Pull request by keerthi-bl**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
> * Did you add or update any new dependencies that are required for your change?

No

Issue: #2135 

Can be tested using Rest client - Postman. 

Sample URL  : http://localhost:8080/resource/api/projects/16f6d03ada424dea81c86fabdab3ee35/summaryAdministration

Sample response : 

{
    "name": "Product Release 1",
    "description": "Product Release 1",
    "version": "1.0",
    "domain": "Application Software",
    "createdOn": "2023-01-18",
    "businessUnit": "DEPARTMENT",
    "state": "ACTIVE",
    "tag": "product",
    "clearingState": "OPEN",
    "roles": {},
    "ownerAccountingUnit": "",
    "ownerGroup": "",
    "ownerCountry": "",
    "preevaluationDeadline": "",
    "systemTestStart": "",
    "systemTestEnd": "",
    "deliveryStart": "",
    "phaseOutSince": "",
    "enableSvm": false,
    "externalIds": {},
    "additionalData": {},
    "considerReleasesFromExternalList": false,
    "enableVulnerabilitiesDisplay": false,
    "clearingSummary": "",
    "specialRisksOSS": "",
    "generalRisks3rdParty": "",
    "specialRisks3rdParty": "",
    "deliveryChannels": "",
    "remarksAdditionalRequirements": "",
    "clearingRequestId": "CR-1",
    "externalUrls": {},
    "modifiedOn": "2023-01-31",
    "projectType": "PRODUCT",
    "visibility": "BUISNESSUNIT_AND_MODERATORS",
    "_links": {
        "self": {
            "href": "http://localhost:8080/resource/api/projects/16f6d03ada424dea81c86fabdab3ee35"
        }
    },
    "_embedded": {
        "leadArchitect": {
            "email": "",
            "deactivated": false,
            "_links": {
                "self": {
                    "href": "http://localhost:8080/resource/api/users/byid"
                }
            }
        },
        "clearingTeam": {
            "email": "Unknown",
            "deactivated": false,
            "_links": {
                "self": {
                    "href": "http://localhost:8080/resource/api/users/byid/Unknown"
                }
            }
        },
        "securityResponsibles": [
            {
                "email": "",
                "deactivated": false,
                "_links": {
                    "self": {
                        "href": "http://localhost:8080/resource/api/users/byid"
                    }
                }
            }
        ],
        "createdBy": {
            "email": "admin@sw360.org",
            "deactivated": false,
            "fullName": "Test Admin",
            "_links": {
                "self": {
                    "href": "http://localhost:8080/resource/api/users/byid/3a678af8c1ce43daa883cd09dd49927e"
                }
            }
        },
        "licenseInfoHeaderText": "",
        "modifiedBy": {
            "email": "keerthi.bl@sw360.org",
            "deactivated": false,
            "fullName": "keerthi bl",
            "_links": {
                "self": {
                    "href": "http://localhost:8080/resource/api/users/byid/7312f989631847e2951cb12a9387ad72"
                }
            }
        },
        "projectOwner": {
            "email": "",
            "deactivated": false,
            "_links": {
                "self": {
                    "href": "http://localhost:8080/resource/api/users/byid"
                }
            }
        },
        "projectResponsible": {
            "email": "",
            "deactivated": false,
            "_links": {
                "self": {
                    "href": "http://localhost:8080/resource/api/users/byid"
                }
            }
        }
    }
}

> Have you implemented any additional tests?
No

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

**Comment by nikkuma7** [comment]
Test successful.
image

File: rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ProjectSpecTest.java [hunk 1]
```diff
@@ -1959,15 +1959,14 @@ public void should_document_create_summary_administration() throws Exception {
...
-                                fieldWithPath("securityResponsibles").description("An array of users responsible for security of the project."),
-                                fieldWithPath("projectResponsible").description("A user who is responsible for the project."),
                                 fieldWithPath("enableSvm").description("Security vulnerability monitoring flag"),
                                 fieldWithPath("considerReleasesFromExternalList").description("Consider list of releases from existing external list"),
                                 fieldWithPath("enableVulnerabilitiesDisplay").description("Displaying vulnerabilities flag."),
                                 fieldWithPath("state").description("The project active status, possible values are: " + Arrays.asList(ProjectState.values())),
                                 fieldWithPath("phaseOutSince").description("The project phase-out date"),
                                 fieldWithPath("clearingRequestId").description("Clearing Request id associated with project."),
                                 fieldWithPath("licenseInfoHeaderText").description("Display licenseInfoHeaderText info"),
```
Comment (rudra-superrr) [inline rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ProjectSpecTest.java:1968]: Remove line number 1968 as you already mentioned licenseInfoHeadertext details on line 1969.

Comment (rudra-superrr) [reply rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ProjectSpecTest.java:1968]: Also add fields like securityResponsibles and projectResponsoble as they are present in response structure.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java [hunk 2]
```diff
@@ -1870,12 +1870,52 @@ public ResponseEntity<EntityModel<Project>> getAdministration(
...
+        sw360Project.unsetSecurityResponsibles();
+        sw360Project.unsetProjectResponsible();
 
         return new ResponseEntity<>(userHalResource, HttpStatus.OK);
     }
 
+    private void setAdditionalFieldsToHalResource(Project sw360Project, HalResource<Project> userHalResource) throws TException {
+        try {
+            String licenseInfoHeaderText = sw360Project.getLicenseInfoHeaderText();
+            if(licenseInfoHeaderText == null || licenseInfoHeaderText.equals(null)) {
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java:1887]: Use proper coding standard for if condition.

**Comment by rudra-superrr** [comment]
Following errors found while testing:
1. ProjectResponsible and securityResponsibles coming twice in response structure. 
image
image
image

2. SecurityResponsibles field is empty.
3. LicenseHeaderText is also empty.

**Comment by keerthi-bl** [comment]
**LicenseHeaderText** will be taking as a different issue, since it will be captured based on groups it will be written as a separate API. The issue reference for the same will be : 
https://code.siemens.com/sw360/sw360portal/-/issues/1376 (internal issue reference).

**Comment by keerthi-bl** [comment]
Review comments addressed.

**Comment by rudra-superrr** [comment]
Code looks good.

**Pull request merged**